overlord/configstate/configcore/watchdog_test.go

```diff
@@ -0,0 +1,129 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/overlord/configstate/configcore"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type watchdogSuite struct {
+	configcoreSuite
+
+	mockEtcEnvironment string
+}
+
+var _ = Suite(&watchdogSuite{})
+
+func (s *watchdogSuite) SetUpTest(c *C) {
+	s.configcoreSuite.SetUpTest(c)
+
+	dirs.SetRootDir(c.MkDir())
+	err := os.MkdirAll(filepath.Join(dirs.GlobalRootDir,
+		"/etc/systemd/system.conf.d/"), 0755)
+	c.Assert(err, IsNil)
+	s.mockEtcEnvironment = filepath.Join(dirs.GlobalRootDir,
+		"/etc/systemd/system.conf.d/10-watchdog.conf")
+}
+
+func (s *watchdogSuite) TearDownTest(c *C) {
+	dirs.SetRootDir("/")
+}
+
+func (s *watchdogSuite) makeMockEtcEnvironment(c *C) {
```

Comment:
AFAICT you don't need this helper because the watchdog code will create an empty file if the `10-watchdog.conf` file is missing and a new tmpdir is used in every tests below. 

It can be useful to test the case where there is an existing file and the test would check that it really gets replaced with the new content. But then I think makeMockEtcEnvironment should write some canary into it (like `# canary content`) and then checking that the content is no longer there after the file got rewriten.